lib/basevm.py

```diff
@@ -69,6 +70,10 @@ def destroy(self):
             self.destroy_vdi(vdi_uuid)
         self._destroy()
 
+    # @abstractmethod
+    def _destroy(self) -> None:
+        raise NotImplementedError()
+
```

Comment:
@stormi about https://github.com/xcp-ng/xcp-ng-tests/pull/284/commits/0fa9a461c80c7bd18644fa31e73bd1e829ddd4cd ...

Double checked what happens when calling `vm-destroy` on a clone and on a snapshot: both leave their VDI behind.  What could be the reason to have/allow different `_destroy` implementations then?  Shouldn't we just fold this `_destroy` into `destroy`?

Reply:
Given the fixme above method `destroy`, I think the only reason why there's an implementation solely for VM is that at the time I needed it for VMs, knew that there were some differences between VMs and snapshots but not which exactly, and so implemented it just for VMs.